### DISCUSSION TITLE
How to create a new Mol instance from molecule structure ?

### DISCUSSION THREAD

**Discussion by gVallverdu** [rdkit/rdkit#5341] [Q&A]
Hi

Looking at the `Chem` module, I can get a `Mol` object from various ways including mol or PDB files (using for example `Chem.MolFromMolFile`). I wonder if there is a way to set up this `Mol` object from the same information as contained in such files but using basic python objects such as list, arrays or others.

Assume you have got such a mol file:

```/User/folder/ethanal.xyz
__Jmol-14_05302215503D 1   1.00000     0.00000     0
Jmol version 14.31.23  2021-01-04 12:16 EXTRACT: ({0:6})
  7  6  0  0  0  0            999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0
    0.8660    0.8660    0.8660 O   0  0  0  0  0  0
    0.2458   -1.0637   -0.1347 H   0  0  0  0  0  0
   -1.2012    0.5845   -0.6824 C   0  0  0  0  0  0
   -2.1065    0.0579   -0.3459 H   0  0  0  0  0  0
   -1.0951    0.4721   -1.7715 H   0  0  0  0  0  0
   -1.2829    1.6521   -0.4302 H   0  0  0  0  0  0
  1  3  1  0  0  0
  1  2  1  0  0  0
  4  6  1  0  0  0
  4  1  1  0  0  0
  5  4  1  0  0  0
  7  4  1  0  0  0
M  END
```

My question is, how can I set up a `Mol` object from, for example:

```py
# a list of elements
['C', 'O', 'H', 'C', 'H', 'H', 'H']
# their coordinates
array([[ 0.     ,  0.     ,  0.     ],
       [ 0.86603,  0.86603,  0.86603],
       [ 0.24576, -1.0637 , -0.13469],
       [-1.20116,  0.58448, -0.68236],
       [-2.10645,  0.05793, -0.34591],
       [-1.09509,  0.47211, -1.77145],
       [-1.2829 ,  1.65205, -0.43017]])
# connectivity 
[(0, 1), (0, 2), (0, 3), (3, 4), (3, 5), (3, 6)]
```

I can obviously export that information to a `.mol` file and read it from rdkit using such like `Chem.MolFromMolFile`. However, I would like to avoid this step of saving data to a file.

I tried to do it by first creating atom objects (`Chem.Atom`), but then I did not see how to set coordinates associated to that atom.

Thank you for any help

**Comment by wopozka** [comment]
For creation of molecule you need editablemol class, for coordinates Conformer class
https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html
